moja-kuhinjica-frontend/src/components/cart/CartItem.tsx

```diff
@@ -1,18 +1,28 @@
-import React from 'react'
+import React, { useState } from 'react'
 import Image from 'next/image'
 import { Text } from '../label/Text'
 import { AmountButton } from '../button/AmountButton'
 import styles from './CartItem.module.scss'
 import mealPic from 'public/static/assets/images/meal2.png'
 import bin from 'public/static/assets/images/bin.svg'
-
-interface ICartItemProps {
-    item?: {
-        name: string
-        price: number
-    }
+import { IMeal } from '@/service/Restaurant.service'
+import { useAppDispatch, useAppSelector } from '@/utils/hooks'
+import { removeCartItem } from '@/reduxStore/reducers/restaurantReducer'
+interface ICartItemPRops {
+    meal: IMeal
 }
-export const CartItem = ({ item }: ICartItemProps): JSX.Element => {
+export const CartItem = ({ meal }: ICartItemPRops): JSX.Element => {
+    const dispatch = useAppDispatch()
+    const amount = useAppSelector(
+        (state) =>
+            state.restaurant.cartItems.find((item) => item.meal.id == meal.id)
+                ?.amount
+    )
+    const getTotalMealPrice = (): number => {
+        let totalPrice = 0
+        if (amount) totalPrice = meal.price * amount
+        return totalPrice
```

Comment:
`const getTotalMealPrice = (): number => amount ? amount * meal?.price : NIL_MEAL`

Reply:
NIL_MEAL je samo predlog za 0 to znaci u top file `const NIL_MEAL = 0`